plugin/turbojpeg/src/main/java/it/geosolutions/imageio/plugins/turbojpeg/TurboJpegImageReader.java

```diff
@@ -206,24 +198,16 @@ public void setInput(Object input, boolean seekForwardOnly, boolean ignoreMetada
         } catch (Exception ex) {
             throw new RuntimeException("Error creating jpegturbo decompressor: " + ex.getMessage(),
                     ex);
-        } finally {
-
-            if (decompressor != null) {
-                try {
-                    decompressor.close();
-                    decompressor = null;
-                } catch (Exception ex) {
-                    // Eat exception. There is nothing else we can do
-                }
-            }
         }
         super.setInput(input, seekForwardOnly, ignoreMetadata);
         // CHECKME: should we mark the position?
     }
 
     public void reset() {
         super.setInput(null, false, false);
-        dispose();
+        if (data != null) {
```

Comment:
Hum... looking at it with fresh eyes, the if does not actually make sense... can just set to null directly :-)
Oh well, going to fix it locally and then merged manually :)